File: mobile/src/app/api/models/create-expense-dto.ts
/* tslint:disable */
import { ExpenseStatus } from './expense-status';
export interface CreateExpenseDto {
  adnotation?: string;
  amount: number;
  deposit: number;
  expenseStatus: ExpenseStatus;
  name: string;
  plannerId: number;
  serviceProviderId?: number;
}

File: WPA.backend/WPA.backend/DTOs/Budget/Expenses/CreateExpenseDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using WPA.backend.Entities;

namespace WPA.backend.DTOs.Expenses
{
    public class CreateExpenseDto
    {
        [Required]
        public int PlannerId { get; set; }

        public int? ServiceProviderId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public decimal Amount { get; set; }

        public string Adnotation { get; set; }

        [Required]
        public decimal Deposit { get; set; }

        [Required]
        public ExpenseStatus ExpenseStatus { get; set; }
    }
}
